lib/SimpleSAML/Metadata/SAMLParser.php

```diff
@@ -124,6 +124,11 @@ class SimpleSAML_Metadata_SAMLParser
      */
     private $entityAttributes;
 
+    /**
+     * An associative array of attributes from the RegistrationInfo element.
+     * @var array
+     */
+    private $registrationInfo;
```

Comment:
I think this addition to `SimpleSAML_Metadata_SAMLParser` makes a lot of sense by itself, but I feel like it should have its own PR. It is just a prerequisite for the functionality implemented by this PR, but others might find this useful too.

That way we can also make this PR smaller and easier to manage.